azimuth/plots/dataset_warnings.py

```diff
@@ -432,113 +434,69 @@ def class_representation(
 
     Returns:
         Plot for one label.
 
     """
     fig = go.Figure()
 
-    train_count_norm = train_count / max(sum(train_count), 1)
-    eval_count_norm = eval_count / max(sum(eval_count), 1)
+    hist_normalized = {split: hist / max(sum(hist), 1) for split, hist in hist_per_split.items()}
 
     # Helps position the error bars.
-    max_y = max(train_count_norm.max(), eval_count_norm.max())
-
-    train_mean = train_stats[0]
-    train_std = train_stats[1]
-
-    eval_mean = eval_stats[0]
-    eval_std = eval_stats[1]
-
-    if eval_count_norm.max() != 0:
-        fig.add_bar(
-            x=list(range(1, len(eval_count_norm) + 1)),
-            y=eval_count_norm,
-            name="evaluation set",
-            marker=dict(color=Colors.DataViz1),
-        )
-
-        fig.add_scatter(
-            x=[np.round(eval_mean, 2)],
-            y=[max_y * 1.14],
-            name="eval_mean_std",
-            error_x=dict(type="constant", value=np.round(eval_std, 2)),
-            hoverinfo="x",
-            marker=dict(color=Colors.DataViz1),
-        )
-
-    if train_count_norm.max() != 0:
-        fig.add_bar(
-            x=list(range(1, len(train_count_norm) + 1)),
-            y=train_count_norm,
-            name="training set",
-            marker=dict(color=Colors.DataViz2, opacity=0.5),
-        )
+    max_y = max([value.max() for value in hist_normalized.values()])
+
+    y_per_split = {DatasetSplitName.eval: 1.14, DatasetSplitName.train: 1.07}
+    opacity_per_split = {DatasetSplitName.eval: 1, DatasetSplitName.train: 0.5}
+    # reversed because the color overlay looks better when train is first
+    for split, hist in reversed(hist_normalized.items()):
+        if hist.max() != 0:
+            fig.add_bar(
+                x=list(range(1, len(hist) + 1)),
+                y=hist,
+                name=DATASET_SPLIT_PRETTY_NAMES[split],
+                marker=dict(color=DATASET_SPLIT_COLORS[split], opacity=opacity_per_split[split]),
+            )
 
-        fig.add_scatter(
-            x=[np.round(train_mean, 2)],
-            y=[max_y * 1.07],
-            name="train_mean_std",
-            error_x=dict(type="constant", value=np.round(train_std, 2)),
-            hoverinfo="x",
-            marker=dict(color=Colors.DataViz2),
-        )
+            fig.add_scatter(
+                x=[np.round(value_per_agg_per_split[split][Agg.mean], 2)],
+                y=[max_y * y_per_split[split]],
+                name=f"{split}_mean_std",
+                error_x=dict(
+                    type="constant", value=np.round(value_per_agg_per_split[split][Agg.std], 2)
+                ),
+                hoverinfo="x",
+                marker=dict(color=DATASET_SPLIT_COLORS[split]),
+            )
 
-    at_least_one_distribution = min(train_count_norm.max(), eval_count_norm.max()) > 0
+    at_least_one_distribution = min([value.max() for value in hist_normalized.values()]) > 0
```

Comment:
In practice, you wouldn't see a real difference in performance, but here we could use `any` and move the check `> 0` inside the generator, so that it breaks our of the loop as soon as it finds one, instead of generating the whole list every time.
```suggestion
    at_least_one_distribution = any(value > 0 for h in hist_normalized.values() for value in h)
```